packages/language-core/lib/codegen/template/element.ts

```diff
@@ -229,7 +229,7 @@ export function* generateComponent(
 		ctx.templateRefs.set(refName, [varName, offset!]);
 		ctx.usedComponentCtxVars.add(var_defineComponentCtx);
 
-		yield `var ${varName} = {} as (Parameters<typeof ${var_defineComponentCtx}['expose']>[0] | null)`;
+		yield `var ${varName} = {} as (Parameters<Required<typeof ${var_defineComponentCtx}>['expose']>[0] | null)`;
```

Comment:
```suggestion
		yield `var ${varName} = {} as (Parameters<NonNullable<typeof ${var_defineComponentCtx}['expose']>>[0] | null)`;
```

I guess this way is more efficient.